Intersect.Server.Core/Entities/Events/CommandProcessing.cs

```diff
@@ -1546,6 +1546,48 @@ Stack<CommandInstance> callStack
             PacketSender.SendEntityDataToProximity(player);
         }
 
+        // Cast Spell On command
+        private static void ProcessCommand(
+           CastSpellOn command,
+           Player player,
+           Event instance,
+           CommandInstance stackInfo,
+           Stack<CommandInstance> callStack
+       )
+        {
+            if (player == null || !player.Online)
+            {
+                return;
+            }
+
+            List<Player> affectedPlayers = new List<Player>();
+
+            if (command.Self)
+            {
+                affectedPlayers.Add(player);
+            }
+
+            if (command.PartyMembers && player.IsInParty)
+            {
+                affectedPlayers.AddRange(player.Party.Where(pl => pl.Id != player.Id));
+            }
+
+            if (command.GuildMembers && player.IsInGuild)
+            {
+                affectedPlayers.AddRange(player.Guild.FindOnlineMembers().Where(pl => pl.Id != player.Id));
```

Comment:
`DistinctBy` is necessary for the overlap between the party/guild, so you don't need to filter out `player.Id` here since it will be taken care of by `DistinctBy`

Reply:
I have this here in the (unlikely) event someone wants to perform a spell on guild members but _not_ themselves or their party. I think it's still necessary

Reply:
Oh yeah, good point.